github-event-service/src/push.ts

```diff
@@ -0,0 +1,59 @@
+import * as fetch from 'node-fetch';
+import {safeLoad as parseYaml} from 'js-yaml';
+
+import {getFirebaseInstance} from './firebase-init';
+
+/** The settings object for the branch manager. */
+export interface BranchManagerSettings {
+  enabled: boolean;
+  branches: Array<{
+    branch: string;
+    label: string;
+  }>;
+};
+
+/** 
+ * Limited implementation of the Github push webhook event from
+ * https://developer.github.com/webhooks/#payloads
+ */
+export interface GithubPushEvent { 
+  ref: string;
+  repository: {
+    id: number;
+    full_name: string;
+  }
+};
+
+/** The Firebase app. */
+const firestore = getFirebaseInstance().firestore();
+
+/** Handle webhook events from github of the type, `push`. */
+export function handlePushEvent(event: GithubPushEvent) {
+  if (event.ref === 'refs/tags/master') {
+    updateRepoSettings(event);
+  }
+  // TODO(josephperrott): Handle pushes to non-master branches.
+}
+
+/** 
+ * Update the settings object in firestore for the repo from the event if the
+ * settings have changed.
+ */
+export async function updateRepoSettings (event: GithubPushEvent) {
```

Comment:
Since the settings aren't passed in, `updateRepoSettingsFromYaml` or `persistRepoSettingsFromSource`? Or maybe just `persistRepoSettings`?

Reply:
how do you feel about `resyncRepoConfig`?

Reply:
The main thing for me is the "FromXxx" part